### ISSUE TITLE
Logger captures warnings from outside PINT

### ISSUE THREAD

**Issue by rossjjennings** [nanograv/PINT#1233]
The `loguru` logger currently captures all warnings when PINT is imported, whether the warnings are generated by PINT or not. This can be a nuisance when it leads to warnings appearing multiple times, or failing to be suppressed when they otherwise would be. A minimal example is the following:

```
import numpy as np
import pint

for i in range(10):
    np.mean(np.array([]))
```

This outputs
```
WARNING  (pint.logging                  ): <class 'RuntimeWarning'> Mean of empty slice.
/home/ross/anaconda3/lib/python3.9/site-packages/numpy/core/_methods.py:169: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning, stacklevel=2)
/home/ross/anaconda3/lib/python3.9/site-packages/numpy/core/_methods.py:189: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
WARNING  (pint.logging                  ): <class 'RuntimeWarning'> Mean of empty slice.
WARNING  (pint.logging                  ): <class 'RuntimeWarning'> Mean of empty slice.
WARNING  (pint.logging                  ): <class 'RuntimeWarning'> Mean of empty slice.
WARNING  (pint.logging                  ): <class 'RuntimeWarning'> Mean of empty slice.
WARNING  (pint.logging                  ): <class 'RuntimeWarning'> Mean of empty slice.
WARNING  (pint.logging                  ): <class 'RuntimeWarning'> Mean of empty slice.
WARNING  (pint.logging                  ): <class 'RuntimeWarning'> Mean of empty slice.
WARNING  (pint.logging                  ): <class 'RuntimeWarning'> Mean of empty slice.
WARNING  (pint.logging                  ): <class 'RuntimeWarning'> Mean of empty slice.
```

**Comment by rossjjennings** [comment]
I tested this in a Jupyter notebook. Running
```
import warnings
print(warnings.filters)
```
in the same notebook gives the following:
```
[('always', None, <class 'scipy.special._sf_error.SpecialFunctionWarning'>, None, 0), ('ignore', re.compile('the matrix subclass is not the recommended way', re.IGNORECASE), <class 'Warning'>, None, 0), ('always', None, <class 'astropy.table.column.StringTruncateWarning'>, None, 0), ('ignore', re.compile('numpy.ndarray size changed', re.IGNORECASE), <class 'Warning'>, None, 0), ('ignore', re.compile('numpy.ufunc size changed', re.IGNORECASE), <class 'Warning'>, None, 0), ('ignore', re.compile('numpy.dtype size changed', re.IGNORECASE), <class 'Warning'>, None, 0), ('always', None, <class 'numpy.RankWarning'>, None, 0), ('error', None, <class 'IPython.core.completer.ProvisionalCompleterWarning'>, None, 0), ('default', None, <class 'DeprecationWarning'>, '__main__', 0), ('ignore', None, <class 'DeprecationWarning'>, None, 0), ('ignore', None, <class 'PendingDeprecationWarning'>, None, 0), ('ignore', None, <class 'ImportWarning'>, None, 0), ('ignore', None, <class 'ResourceWarning'>, None, 0)]
```

**Comment by dlakaplan** [comment]
I think capturing warnings from outside PINT is intended behavior (e.g., various matplotlib issues).  the problem is that it may not be following existing filters.  When I run this on my own I see that warning every time, inside PINT or otherwise.  

What happens if you just do it in a plain python shell where you haven't imported PINT?

And what happens in a notebook where you haven't imported PINT?

I don't see an entry for a `RuntimeWarning` in your warnings filter, so I'm not sure what's suppressing it.

**Comment by dlakaplan** [comment]
This should already work:
```
warnings.filterwarnings(
    "ignore",
    category=RuntimeWarning,
)
```
But it won't work with the `"once"` option.

**Comment by rossjjennings** [comment]
For me, it doesn't seem to matter whether I'm running from a plain python shell, a script, or a notebook. The output is the same. When I haven't imported PINT, I just get
```
/home/ross/anaconda3/lib/python3.9/site-packages/numpy/core/_methods.py:169: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning, stacklevel=2)
/home/ross/anaconda3/lib/python3.9/site-packages/numpy/core/_methods.py:189: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
```

**Comment by dlakaplan** [comment]
But do you get that one time or many?

**Comment by rossjjennings** [comment]
Just once.

**Comment by rossjjennings** [comment]
As a user, I guess I expect PINT logging to pick up errors generated in other code (whether that's numpy, matplotlib, ERFA, or whatever else) only if they were generated by a call to that code within PINT. In this example code (and in the more complicated case where this originally came up), I'm doing something that's not related to PINT at all, except for the fact that I happen to have imported it, so it seems strange to me that the PINT logger inserts itself into things at all. It makes it seem like the warnings came from PINT, when they really didn't.

**Comment by dlakaplan** [comment]
There's really no way to know who issues a warning - related to PINT or not.  If we want to leave all other warnings alone that's easy enough to change but there are other reasons for not doing that.   We can discuss that more broadly, or see if there is a good way to make it configurable.

The thing that I don't like is that it obscures the origin.  I have a partial fix for that, and to make it respect `"once"` filters.

**Comment by dlakaplan** [comment]
Fixed by #1232
